src/map/lua/lua_baseentity.cpp

```diff
@@ -8625,6 +8625,61 @@ inline int32 CLuaBaseEntity::getAllianceSize(lua_State* L)
     return 1;
 }
 
+/************************************************************************
+*  Function: getAllianceID()
+*  Purpose : Returns the id of the alliance
+*  Example : local allianceId = player:getAllianceID()
+*  Notes   :
+************************************************************************/
+
+inline int32 CLuaBaseEntity::getAllianceID(lua_State* L)
+{
+    DSP_DEBUG_BREAK_IF(m_PBaseEntity == nullptr);
+    DSP_DEBUG_BREAK_IF(m_PBaseEntity->objtype != TYPE_PC);
+
+    uint8 id = 0;
+
+    if (const CCharEntity* PChar = dynamic_cast<CCharEntity*>(m_PBaseEntity))
+    {
+        if (PChar->PParty)
+        {
+            if (PChar->PParty->m_PAlliance)
+                id = PChar->PParty->m_PAlliance->m_AllianceID;
+            lua_pushnumber(L, id);
+            return 1;
+        }
+    }
+    lua_pushnil(L);
+    return 1;
+}
+
+/************************************************************************
+*  Function: getPartyID()
+*  Purpose : Returns the id of the party
+*  Example : local partyId = player:getPartyID()
+*  Notes   :
+************************************************************************/
+
+inline int32 CLuaBaseEntity::getPartyID(lua_State* L)
+{
+    DSP_DEBUG_BREAK_IF(m_PBaseEntity == nullptr);
+    DSP_DEBUG_BREAK_IF(m_PBaseEntity->objtype != TYPE_PC);
+
+    uint8 id = 0;
+
+    if (const CCharEntity* PChar = dynamic_cast<CCharEntity*>(m_PBaseEntity))
+    {
+        if (PChar->PParty)
+        {
+            id = PChar->PParty->GetPartyID();
```

Comment:
may as well ditch the variable at this point and just move that right into the lua_puchnumber line

Reply:
you know i thought of doing that, wish i did now

Reply:
Done